packages/utils/src/validation.ts

```
@@ -16,28 +16,35 @@ function validateChemJson(obj: IChemJson) : boolean {
     // If bonds are invalid, throw them out but still keep the atoms
     if (!validateBonds(obj.atoms, obj.bonds)) {
       console.warn('Invalid CJSON data, discarding bonds');
-      obj.bonds = undefined;
+      delete obj.bonds;
     }
   }
   if (!isNil(obj.cube)) {
     // If cube data is invalid, throw it out but still keep the rest
     if (!validateCube(obj.cube)) {
       console.warn('Invalid CJSON data, discarding cube');
-      obj.cube = undefined;
+      delete obj.cube;
```

Isn't this going to silently fail? As in I am trying to add a cube, but the format is wrong so its delete, if I don't have the console open I would non the wiser?

---

It will silently fail, but at least the portions of the cjson that are properly formatted will still be displayed.

What behavior would you suggest?

---

Throw an exception?